va/va_trace.c

```diff
@@ -6344,6 +6334,16 @@ void va_TraceExportSurfaceHandle(
         return;
     }
 
+    int i;
+
+    DPY2TRACE_VIRCTX(dpy);
+
+    TRACE_FUNCNAME(idx);
+
+    va_TraceMsg(trace_ctx, "\tsurfaceId = 0x%08x\n", surfaceId);
```

Comment:
I liked that these 3 items would be printed regardless of memType check.... Consider that someone will need to extend this function to cover other mem types - he will need to basically revert your change.

Thus, please, revert back the change and handle mutex issue in a different way, i.e. either call `DPY2TRACE_VIRCTX_EXIT` before each return or use `goto exit` instead of return.

Reply:
Thanks. Updated.